### PR TITLE
[UR][CUDA] Fix function pointer typedefs in xpti tracing

### PR DESCRIPTION
**Pull request by konradkusiak97**
CI for UR PR: https://github.com/oneapi-src/unified-runtime/pull/1368

### CODE REVIEW

**Comment by konradkusiak97** [comment]
Doesn't seem like the CI failures are related to this fix

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 1]
```diff
@@ -56,14 +56,8 @@ endif()
...
-  set(UNIFIED_RUNTIME_REPO "https://github.com/oneapi-src/unified-runtime.git")
-  # commit e2ee9a4720414e0a59fa9c911e9575ab564ac57c
-  # Merge: 7a5150cd be622e7c
-  # Author: Kenneth Benzie (Benie) <k.benzie@codeplay.com>
-  # Date:   Sun Mar 10 18:02:50 2024 +0000
-  #     Merge pull request #1340 from Bensuo/ewan/coverity_cuda_update
-  #     [HIP][CUDA][Command-Buffer] Fix Coverity issues in HIP/CUDA command-buffer code
-  set(UNIFIED_RUNTIME_TAG e2ee9a4720414e0a59fa9c911e9575ab564ac57c)
+  set(UNIFIED_RUNTIME_REPO "https://github.com/konradkusiak97/unified-runtime.git")
+  set(UNIFIED_RUNTIME_TAG 0446c65e869d5098fa006d0035a5b00240400bee)
```
Comment (kbenzie) [inline sycl/plugins/unified_runtime/CMakeLists.txt:60]:
```suggestion
  set(UNIFIED_RUNTIME_REPO "https://github.com/oneapi-src/unified-runtime.git")
  # commit 1cf9a08e6a98381fadbc065de2700b463d2a44a4
  # Merge: c49b116f 0446c65e
  # Author: Kenneth Benzie (Benie) <k.benzie@codeplay.com>
  # Date:   Wed Mar 13 09:48:36 2024 +0000
  #     Merge pull request #1368 from konradkusiak97/xptiWindowsFix
  #     [CUDA][XPTI] Fix function pointer typedefs in xpti tracing
  set(UNIFIED_RUNTIME_TAG 1cf9a08e6a98381fadbc065de2700b463d2a44a4)
```

**Review by kbenzie — COMMENTED** [review]
https://github.com/oneapi-src/unified-runtime/pull/1368 has been merged, however this PR is now dependent https://github.com/intel/llvm/pull/12833 merging first. Once the dependent PR has merged please merge in the latest sycl branch changes, update the UR repo/tag, then make this PR ready for review.

File: sycl/plugins/unified_runtime/CMakeLists.txt (hunk 1 above)
Comment (konradkusiak97) [reply sycl/plugins/unified_runtime/CMakeLists.txt]: Done

Comment (kbenzie) [reply sycl/plugins/unified_runtime/CMakeLists.txt]: https://github.com/intel/llvm/pull/12833 has merged now, go ahead an fix the conflict and make this ready for review, then I'll approve it.

**Comment by kbenzie** [comment]
@intel/llvm-gatekeepers please merge

**Pull request merged**